File: TOMSMorphingLabelExample/TOMSMorphingLabelExample/ViewController.m
//
//  ViewController.m
//  TOMSMorphingLabelExample
//
//  Created by Tom König on 13/06/14.
//  Copyright (c) 2014 TomKnig. All rights reserved.
//

#import "ViewController.h"
#import <TOMSMorphingLabel.h>

@interface ViewController ()
@property (weak, nonatomic) IBOutlet TOMSMorphingLabel *morphingLabel;
@property (nonatomic, assign) NSInteger idx;
@property (nonatomic, strong) NSArray *textValues;
@end

@implementation ViewController
            
- (void)viewDidLoad {
    [super viewDidLoad];
    
    _idx = 0;
    [self toggleText];
}

#pragma mark - toggling text

- (NSArray *)textValues
{
    if (!_textValues) {
        _textValues = @[
                        @"Swift",
                        @"Swiftilicious",
                        @"delicious",
                        @"开",
                        @"开源",
                        @"2⃣3⃣4⃣",
                        @"1⃣2⃣3⃣4⃣5⃣",
                        @""
                        ];
    }
    return _textValues;
}

- (void)setIdx:(NSInteger)idx
{
    _idx = MAX(0, MIN(idx, idx % [self.textValues count]));
}

- (void)toggleText
{
    self.morphingLabel.text = self.textValues[self.idx++];

    [self performSelector:@selector(toggleText)
               withObject:nil
               afterDelay:2];
}

@end

File: TOMSMorphingLabelExample/TOMSMorphingLabelExample/ViewController.swift
//
//  ViewController.swift
//  TOMSMorphingLabelExample
//
//  Created by Tom König on 12/06/14.
//  Copyright (c) 2014 TomKnig. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
                            
    override func viewDidLoad() {
        super.viewDidLoad()
        showMorphingLabel()
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func showMorphingLabel() {
        let textLabelFrame = CGRect(x: 0, y: 42, width: self.view.frame.size.width, height: 42)
        var textLabel = TOMSMorphingLabel(frame: textLabelFrame)
        
        textLabel.font = UIFont.systemFontOfSize(32)
        textLabel.textColor = UIColor(red: 0.102, green: 0.839, blue: 0.992, alpha: 1)
        textLabel.textAlignment = NSTextAlignment.Center
        
        self.view.addSubview(textLabel)
        
        self.insertTestText(textLabel)
    }
    
    func insertTestText(label: UILabel) {
        label.text = "Swift"
        dispatch_after(dispatch_time(DISPATCH_TIME_NOW, Int64(1.0 * Double(NSEC_PER_SEC))), dispatch_get_main_queue(), {
            label.text = "Swiftilicious"
            dispatch_after(dispatch_time(DISPATCH_TIME_NOW, Int64(1.0 * Double(NSEC_PER_SEC))), dispatch_get_main_queue(), {
                label.text = "delicious"
                dispatch_after(dispatch_time(DISPATCH_TIME_NOW, Int64(1.0 * Double(NSEC_PER_SEC))), dispatch_get_main_queue(), {
                    label.text = ""
                    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, Int64(1.0 * Double(NSEC_PER_SEC))), dispatch_get_main_queue(), {
                        self.insertTestText(label)
                        })
                    })
                })
            })
    }


}

